### PR TITLE
Change request advanced search and filter

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Filtering and searching change requests.

filtering options:
Screenshot 2023-08-22 at 16 40 46

searching options:
Screenshot 2023-08-22 at 16 40 53

### CODE REVIEW

File: frontend/src/component/common/Search/SearchSuggestions/SearchSuggestions.tsx [hunk 1]
```diff
@@ -69,16 +71,19 @@ export const SearchSuggestions: VFC<SearchSuggestionsProps> = ({
...
                 header: column.Header ?? column.filterName,
-                options: [...new Set(filterOptions)]
-                    .filter(Boolean)
-                    .flatMap(item => item.split('\n'))
-                    .map(item => (item.includes(' ') ? `"${item}"` : item))
-                    .sort((a, b) => a.localeCompare(b)),
+                options,
                 suggestedOption:
-                    filterOptions[randomRow] ?? `example-${column.filterName}`,
+                    options[randomRow] ?? `example-${column.filterName}`,
```
Comment (kwasniew) [inline frontend/src/component/common/Search/SearchSuggestions/SearchSuggestions.tsx:86]: we want to suggest from the valid options after pre-processing. E.g. with quotes, with split new line

File: frontend/src/component/common/Search/SearchSuggestions/SearchSuggestions.tsx [hunk 2]
```diff
@@ -69,16 +71,19 @@ export const SearchSuggestions: VFC<SearchSuggestionsProps> = ({
                 getColumnValues(column, row)
             );
 
+            const options = [...new Set(filterOptions)]
+                .filter(Boolean)
+                .flatMap(item => item.split('\n'))
+                .filter(item => !item.includes('"') && !item.includes("'"))
```
Comment (kwasniew) [inline frontend/src/component/common/Search/SearchSuggestions/SearchSuggestions.tsx:77]: if your field value is using quotes (corner case) we filter it out to not mess with the search quotes

File: frontend/src/component/common/Search/SearchSuggestions/SearchDescription/SearchDescription.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ export const SearchDescription: VFC<ISearchDescriptionProps> = ({
                                     {filter.values.join(',')}
                                 </StyledCode>{' '}
                                 in {filter.header}. Options:{' '}
-                                {filter.options.join(', ')}
+                                {filter.options.slice(0, 10).join(', ')}
```
Comment (kwasniew) [inline frontend/src/component/common/Search/SearchSuggestions/SearchDescription/SearchDescription.tsx:64]: we don't want to present more than 10 options. E.g. for feature names in change request list

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**